src/API/WorkController.ts

```diff
@@ -26,9 +26,11 @@ export class WorkController {
     return this.collection.findOne({ id }, { fields: { _id: false } })
   }
 
-  async getByPublicKey(publicKey: string): Promise<any> {
+  async getByPublicKey(publicKey?: string): Promise<any> {
     this.logger.trace({ method: 'getByPublicKey', publicKey }, 'Getting Works by Public Key from DB')
-    return this.collection.find({ publicKey }, { fields: { _id: false } }).toArray()
+    return publicKey
+      ? this.collection.find({ publicKey }, { fields: { _id: false } }).toArray()
```

Comment:
This isn't going to scale. As we add another param we'll have to add all possible combinations.

Maybe we can do something like 

```ts
interface WorksFilters {
  readonly publicKey?: string
}

async getByFilters(worksFilters: WorksFilters = {}): Promise<any> {
  const definedFilters = Object.entries(worksFilters)
    .filter(([key, value]) => value !== undefined) 
    .toObject()
  return this.collection.find(definedFilters, { fields: { _id: false } }).toArray()
}
```

We can use `undefined` as a special case since MongoDB doesn't allow comparing against `undefined` anyway.

```
> db.works.find({"noField": undefined})
Error: error: {
	"ok" : 0,
	"errmsg" : "cannot compare to undefined",
	"code" : 2,
	"codeName" : "BadValue"
}
```

Reply:
Ok I made these changes. I don't think I'm understanding why we would need to check for undefined though. Isn't Joi only allowing us to pass in known parameters that aren't empty? 
```js
async getByFilters(worksFilters: WorksFilters = {}): Promise<any> {
  return this.collection.find(worksFilters, { fields: { _id: false } }).toArray()
}
```
`Array.toObject` I may have implemented differently than you were expecting as well. Let me know if I need to rework it.

Reply:
I'm not sure either now. Dunno what I was thinking originally...